mlir/lib/Transforms/Utils/DialectConversion.cpp

```diff
@@ -915,6 +918,11 @@ struct ConversionPatternRewriterImpl : public RewriterBase::Listener {
   /// Return "true" if the given operation was replaced or erased.
   bool wasOpReplaced(Operation *op) const;
 
+  /// Set the operation legalizer to use for recursive legalization.
+  void setOperationLegalizer(OperationLegalizer *legalizer) {
```

Comment:
Let me know if you see a better way to layer this. Long-term, we may want to merge `ConversionPatternRewriterImpl` and `OperationLegalizer` into a single class.